From d96ba8d0f2213c302d13388c18a0bbc086fe8d14 Mon Sep 17 00:00:00 2001
From: Matteo Bertozzi <matteo.bertozzi@cloudera.com>
Date: Thu, 17 Dec 2015 09:31:04 -0800
Subject: [PATCH 321/335] HBASE-14730 region server needs to log warnings when
 there are attributes configured for cells with
 hfile v2 (huaxiang sun)

Change-Id: I244b88e69256e1b855e834548f551ab88f3b1447
Reason: Improvement
Author: Huaxiang Sun
Ref: CDH-40134
---
 .../hadoop/hbase/io/hfile/HFileWriterV2.java       |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterV2.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterV2.java
index 28c4655..edab0dc 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterV2.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterV2.java
@@ -88,6 +88,9 @@ public class HFileWriterV2 extends AbstractHFileWriter {
 
   protected long maxMemstoreTS = 0;
 
+  /** warn on cell with tags */
+  private static boolean warnCellWithTags = true;
+
   static class WriterFactoryV2 extends HFile.WriterFactory {
     WriterFactoryV2(Configuration conf, CacheConfig cacheConf) {
       super(conf, cacheConf);
@@ -267,6 +270,13 @@ public class HFileWriterV2 extends AbstractHFileWriter {
       newBlock();
     }
 
+    if (warnCellWithTags && getFileContext().isIncludesTags()) {
+      LOG.warn("A minimum HFile version of " + HFile.MIN_FORMAT_VERSION_WITH_TAGS
+          + " is required to support cell attributes/tags. Consider setting "
+          + HFile.FORMAT_VERSION_KEY + " accordingly.");
+      warnCellWithTags = false;
+    }
+
     fsBlockWriter.write(cell);
 
     totalKeyLength += CellUtil.estimatedSerializedSizeOfKey(cell);
-- 
1.7.9.5

